readthedocs/search/api.py

```diff
@@ -24,10 +63,27 @@ def get_queryset(self):
         # Validate all the required params are there
         self.validate_query_params()
         query = self.request.query_params.get('q', '')
-        queryset = PageDocument.simple_search(query=query)
+        kwargs = {}
+        kwargs['projects_list'] = [p.slug for p in self.get_all_projects()]
+        kwargs['versions_list'] = self.request.query_params.get('version')
+        user = ''
```

Comment:
I think it's a better pattern to default to the `AnonymousUser` instead. So, anywhere where this is used, all the method are still available and returning the proper values.

I think you can pass `self.request.user` directly without checking anything.

Reply:
I agree here, unless there is some significance to ES handling the user as an empty string.

Reply:
Nope, was just the old way we were doing it. Fixed it now.